components/server/src/user/user-controller.ts

```diff
@@ -705,4 +769,17 @@ export class UserController {
         log.debug({ sessionId: req.sessionID }, "The redirect URL does not match", { query: req.query });
         return;
     }
+
+    private createGitpodServer(user: User) {
+        const server = this.serverFactory();
+        server.initialize(
+            undefined,
+            user,
+            new OwnerResourceGuard(user.id),
```

Comment:
Note: This line ensures that only the owner can access the `sendHeartbeat` method for their instances.

Reply:
ouch, it is a valid point in case of live sharing cc @mustard-mh we need to check how construct proper guard similar how we do for web socket connections, i think there should be a service fro HTTP connections as well already